File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_42_incr_list.R
# Return list with elements incremented by 1.
# >>> incr_list([1, 2, 3])
# [2, 3, 4]
# >>> incr_list([5, 3, 5, 2, 3, 3, 9, 0, 123])
# [6, 4, 6, 3, 4, 4, 10, 1, 124]
incr_list <- function(l) {
    l <- l + 1
    return(l)
}
incr_list([1, 2, 3])
incr_list([5, 3, 5, 2, 3, 3, 9, 0, 123])


test_humaneval <- function() {
candidate <- incr_list
    if(!identical(candidate(list()), list())){quit('no', 1)}
    if(!identical(candidate(list(3, 2, 1)), list(4, 3, 2))){quit('no', 1)}
    if(!identical(candidate(list(5, 2, 5, 2, 3, 3, 9, 0, 123)), list(6, 3, 6, 3, 4, 4, 10, 1, 124))){quit('no', 1)}
}
test_humaneval()

File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_42_incr_list.pl
# Return list with elements incremented by 1.
# >>> incr_list([1, 2, 3])
# [2, 3, 4]
# >>> incr_list([5, 3, 5, 2, 3, 3, 9, 0, 123])
# [6, 4, 6, 3, 4, 4, 10, 1, 124]
sub incr_list {
    my($l) = @_;
    my @l = @$l;
    my @r;
    for my $e (@l) {
        push @r, $e + 1;
    }
    return \@r;
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&incr_list;
        if(eq_deeply($candidate->([]),[])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->([3, 2, 1]),[4, 3, 2])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->([5, 2, 5, 2, 3, 3, 9, 0, 123]),[6, 3, 6, 3, 4, 4, 10, 1, 124])) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();